lib/logstash/inputs/zeromq.rb

```diff
@@ -69,7 +69,10 @@ class LogStash::Inputs::ZeroMQ < LogStash::Inputs::Base
   #  * `ZMQ::SWAP_SIZE` - space for disk overflow
   #
   # example: `sockopt => ["ZMQ::HWM", 50, "ZMQ::IDENTITY", "my_named_queue"]`
-  config :sockopt, :validate => :hash
+  #
+  # defaults to: `sockopt => ["ZMQ::RCVTIMEO", "1000"]`, which has the effect of "interrupting"
+  # the recv operation at least once every second to allow for properly shutdown handling.
+  config :sockopt, :validate => :hash, :default => [ "ZMQ::RCVTIMEO", "1000" ]
```

Comment:
I agree, it should be a hash. I just followed the existing example, which provides an array with an even count of elements, which gets converted to a hash in https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/config/mixin.rb#L381 (lines 381 - 398). I update the PR, including the existing comment/example.